### PR TITLE
release-notes.txt: add 2023.10 release notes

### PR DESCRIPTION
**Pull request by MrKevinWeiss**
### Contribution description

That time again, please check it out!

### CODE REVIEW

**Comment by MrKevinWeiss** [comment]
ping @RIOT-OS/maintainers

File: release-notes.txt [hunk 1]
```diff
@@ -1,3 +1,491 @@
...
+* gh-actions: remove reporting release tests to Matrix (#19879)
+* tests/gcoap_fileserver: only enable test with GCC (#19870)
+* tests/net/gcoap_fileserver: disable test on CI (#19898)
+* tests/net/gcoap_fileserver: Fix failing nightlies (#19856)
+* tests/pkg/relic: skip CI testing with samr21-xpro and llvm toolchain (#19902)
+
+API Changes
+-----------
+
+* drivers/mtd: use XFA for pointers to defined MTDs (#19465)
```
Comment (benpicco) [inline release-notes.txt:158]: Uh this still made it into the release? I thought it was merged after the hard feature freeze.
Anyway, you then probably want to add the porting guide from the PR to the release notes.

Comment (MrKevinWeiss) [reply release-notes.txt:158]: I can double check... I had to do some hacks in the release note generation.

File: release-notes.txt [hunk 2]
```diff
@@ -1,3 +1,477 @@
...
+Networking
+----------
+
++ coap: add missing Content-Format definitions (#19875)
++ coap: add missing option numbers (#19874)
++ drivers/atwinc15x0: support dynamic scanning and connection to AP (#19387)
++ sys/net/sock: add sock_aux_ttl (#19836)
+* cmds_gnrc_netif: Support enabling/disabling lwIP netifs from
+  gnrc_netif shell (#19972)
+* lwip: bump to v2.2.0 (#19780)
```
Comment (miri64) [inline release-notes.txt:63]: Somewhat fuzzy, but not sure if this rather belongs to "Packages".

File: release-notes.txt (hunk 1 above)
Comment (MrKevinWeiss) [reply release-notes.txt:158]: It is not in there.  Maybe it was merged just after the freeze and my hard-coded timestamp took it.

**Comment by MrKevinWeiss** [comment]
ping

File: release-notes.txt (hunk 1 above)
Comment (benpicco) [reply release-notes.txt]: Uh does the `set-milestones` sub-command of `riot-release-manager` no longer work?

File: release-notes.txt [hunk 3]
```diff
@@ -1,3 +1,475 @@
...
+licensed with a non-viral copyleft license (LGPLv2.1), which allows indirect
+business models around the free open-source software platform provided by RIOT.
+
+
+About this release
+==================
+
+The 2023.10 release includes:
+
+- PSA Crypto API implementation, one step close to a secure IoT!
```
Comment (kaspar030) [inline release-notes.txt:23]:
```suggestion
- PSA Crypto API implementation, one step closer to a secure IoT!
```

File: release-notes.txt [hunk 4]
```diff
@@ -1,3 +1,475 @@
...
+- A bunch of default drivers for boards
+- Improved clang support
+
```
Comment (kaspar030) [inline release-notes.txt:26]:
```suggestion
```

File: release-notes.txt [hunk 5]
```diff
@@ -1,3 +1,475 @@
...
++ cpu/sam0_common/periph: add low-level SDMMC peripheral driver for SDHC (#19760)
++ cpu/efm32/periph: add DAC support for EFM32 Series 1 (VDAC) (#19887)
+* cpu/atmega_common: hook up BAT LOW irq to power bus (#19822)
+* cpu/riscv_common: remove picolibc from blacklisting in CI (#19862)
+* cpu/stm32/periph/eth: Disable hardware checksums (#19952)
+* cpu/stm32: bump cmsis packages version (#19904)
+* cpu/stm32: fix ld script for SRAM4 (#19842)
+* dist/tools/esptools: upgrade ESP32x toolchains to GCC version 12.2 (#19452)
+* sys/psa_crypto: Ed25519 (EdDSA) support (#19954)
+* tree-wide: mixed box of compilation fixes with clang (#19634)
```
Comment (kaspar030) [inline release-notes.txt:104]: this fits probably better somewhere else (bugfixes?).

File: release-notes.txt [hunk 6]
```diff
@@ -1,3 +1,475 @@
...
+
+Build System / Tooling
+----------------------
+
++ compile_and_test_for_boards: Add no-compile flag (#19817)
++ dist/testbed-support: Add openmote board [backport 2023.10] (#19984)
+* dist/tools/jlink: fix DBG_PID assignment (#19960)
+* dist/tools/usb-serial: call ttys.py with its path (#19823)
+* make: COMPILE_COMMANDS_PATH adapt for external apps (#19869)
+
```
Comment (kaspar030) [inline release-notes.txt:137]: sometimes there's two empty lines, sometimes one. please unify.

Comment (kaspar030) [reply release-notes.txt:137]: (probably use two between sections)

**Review by kaspar030 — APPROVED** [review]
ACK.

**Pull request merged**